external-crates/move/crates/move-compiler/src/editions/mod.rs

```diff
@@ -118,6 +118,7 @@ const E2024_ALPHA_FEATURES: &[FeatureGate] = &[
     FeatureGate::BlockLabels,
     FeatureGate::Move2024Paths,
     FeatureGate::MacroFuns,
+    FeatureGate::Move2024Optimizations,
```

Comment:
I'm confused. I don't see this used anywhere?

Also, are we ready to turn on these optimizations for all versions?

Reply:
The freeze change in HLIR is feature gated by this. I have updated the diff to also enable the CFGIR optimization. This was also partially to help @rvantonder verify the versioning work.

We can postpone if you'd like.

Reply:
Just to chip in, nonessential for this to go in now on my account (I can test locally with patches). Deferring to yinz for making sure this is ready™️

Reply:
I also. had it in mind that we would set up a `2024.prerelease` or some such edition for the things actually shipping in the prerelease.